tests/pipeline/nodes/draw/test_poses.py

```diff
@@ -16,39 +16,147 @@
+    return request.param["value"]
+
+
+@pytest.fixture(
+    params=[
+        {"value": [-1, 255, 255]},
+        {"value": [256, 255, 255]},
+    ]
+)
+def invalid_color_range(request):
+    return request.param["value"]
+
+
+@pytest.fixture(
+    params=[
+        {"value": [0, 0, 0, 0]},
+        {"value": [0, 0]},
+    ]
+)
+def invalid_color_tuple_len(request):
+    return request.param["value"]
+
+
+class TestPoses:
+    def test_no_poses(self, draw_poses_config, create_image):
+        draw_poses_node = Node(draw_poses_config)
+
+        original_img = create_image((28, 28, 3))
+        output_img = original_img.copy()
+
+        keypoints = np.empty((0, 17, 2))  # keypoints shape
+        keypoint_scores = np.empty((0, 17))  # keypoint_scores shape
+        keypoint_conns = np.empty((0, 15, 2, 2))  # keypoint_conns shape
+
+        inputs = {
+            "keypoints": keypoints,
+            "keypoint_scores": keypoint_scores,
+            "keypoint_conns": keypoint_conns,
+            "img": output_img,
         }
+
+        draw_poses_node.run(inputs)
+
+        np.testing.assert_equal(original_img, output_img)
+
+    @pytest.mark.parametrize(
+        "config_key", ["keypoint_dot_color", "keypoint_connect_color"]
+    )
+    def test_invalid_color_name(
+        self, config_key, draw_poses_config, invalid_color_name
+    ):
+        draw_poses_config[config_key] = invalid_color_name
+
+        with pytest.raises(ValueError) as excinfo:
+            _ = Node(draw_poses_config)
+        assert f"{config_key} must be one of" in str(excinfo.value)
+
+    @pytest.mark.parametrize(
+        "config_key", ["keypoint_dot_color", "keypoint_connect_color"]
+    )
+    def test_invalid_color_range(
+        self, config_key, draw_poses_config, invalid_color_range
+    ):
+        draw_poses_config[config_key] = invalid_color_range
+
+        with pytest.raises(ValueError) as excinfo:
+            _ = Node(draw_poses_config)
+        assert f"All elements of {config_key} must be between [0.0, 255.0]" in str(
+            excinfo.value
+        )
+
+    @pytest.mark.parametrize(
+        "config_key", ["keypoint_dot_color", "keypoint_connect_color"]
     )
-    return node
+    def test_invalid_color_tuple_len(
+        self, config_key, draw_poses_config, invalid_color_tuple_len
+    ):
+        draw_poses_config[config_key] = invalid_color_tuple_len
 
+        with pytest.raises(ValueError) as excinfo:
+            _ = Node(draw_poses_config)
+        assert "BGR values must be a list of length 3." in str(excinfo.value)
+
+    @mock.patch("cv2.line", wraps=cv2.line)
+    @mock.patch("cv2.circle", wraps=cv2.circle)
+    def test_if_configs_are_used(
+        self, mock_circle, mock_line, draw_poses_config, create_image
+    ):
+        draw_poses_node = Node(draw_poses_config)
+
+        default_keypoint_dot_color = draw_poses_node.keypoint_dot_color
+        default_keypoint_connect_color = draw_poses_node.keypoint_connect_color
```

Comment:
I think it would be better to set a custom color for this test, even something like `[1, 2, 3]` would be fine. Just to ensure that the values from the configs are passed down to the actual draw functions instead of relying on any hardcoded values.

Reply:
Thanks for pointing out, I have changed accordingly.